### DISCUSSION TITLE
Weird Development Ages on Sample Dataset when using pandas 2.0.2

### DISCUSSION THREAD

**Discussion by FlawlessNa** [casact/chainladder-python#439] [General]
Hello,

First of all thanks for this great package! It is extremely useful!

I recently started using the `chainladder` package and first noticed an issue when combined with `pandas `2.0.2.
I'm not entirely sure whether the issue arises from `pandas `itself, or from the triangle creation within `chainladder`. This is why I decided to ask in the discussion section first.

Note: I've ran the script under both Python 3.8 and 3.11 and got the exact same results.

Here is the script I am running to outline the issue (I've tried with a random triangle sampled from "clrd" and observed the same problem as well):
![image](https://github.com/casact/chainladder-python/assets/106719178/34fc4522-b795-49a2-8e3e-7b066005907b)

If I run this script with `pandas` 1.5.3 (similar version used in the `chainladder` documentation), then everything goes smoothly, there are no problems with the output triangle, as below:
![image](https://github.com/casact/chainladder-python/assets/106719178/424e3daf-2426-4032-8a9b-aaef35d367e3)

Now if I run this script with `pandas 2.0.2`, notice what happens to the Development Ages (and similarly, the `data.valuation` dates end up being problematic as well)

![image](https://github.com/casact/chainladder-python/assets/106719178/882165ae-fa97-487f-8f95-60f107e9bf95)

This is effectively a problem because it messes up the `cl.Development()` model fit when combined with various parameters (n_periods, drop_low/high, etc).

I've tried digging to find the root of the discrepancy, and it looks like it first arises in `TriangleBase._development_lag` which is called when using `cl.load_sample.`
I'm not sure of the best solution but I thought somebody might be able to help!

Thanks!!

**Comment by jbogaardt** [comment]
Thanks, I haven't yet run our test suite against pandas>=2.0. Someone also just posted issue #438 referencing the same.  Sounds like there are breaking changes that need to be addressed. This issue gives me a good place to start looking.
